src/IonReader.ts

```diff
@@ -16,23 +16,139 @@ import { Decimal } from "./IonDecimal";
+     * (i.e. `depth() > 0`).  Note that the positions the reader *after* the container that was
+     * stepped out of, but *before* the next value after the container.  One should generally
+     * call [[next]] after invoking this method.
+     *
+     * @throw Error if the reader is not positioned within a container.
+     */
+    stepOut(): void;
+
+    /** Returns the depth of the reader.  This is `0` if the reader is not inside of a container. */
+    depth(): number;
+
+    // TODO implement symboltokens to replace string[] https://github.com/amzn/ion-js/issues/121
+    /**
+     * Returns the annotations for the current value.
+     *
+     * @return the empty array if the there are no annotations or the reader is not positioned on a value.
+     */
+    annotations(): string[];
+
+    /**
+     * Returns the field name of the current value.
+     *
+     * @return `null` if the reader is not positioned on a value or is in a value that has no field name.
+     */
+    fieldName(): string | null;
+
+    /** Returns true if and only if the reader is positioned on a `null` value of any type. */
+    isNull(): boolean;
+
+    /**
+     * Returns the [[IonType]] associated with the current value.
+     *
+     * @return `null` if the reader is not positioned on a value.
+     */
+    type(): IonType | null;
+
+    /**
+     * Returns the current value as a `boolean`.  This is only valid if `type() == IonTypes.BOOL`.
+     *
+     * @return `null` if the current Ion value [[isNull]].
+     *
+     * @throw Error when the reader is not positioned on a `bool` typed value.
+     */
+    booleanValue(): boolean | null;
+
+    /**
+     * Returns the current value as a `Uint8Array`.  This is only valid if `type() == IonTypes.BLOB`
+     * or `type() == IonTypes.CLOB`.
+     *
+     * @return `null` if the current Ion value [[isNull]].
+     *
+     * @throw Error when the reader is not positioned on a `clob` or `blob` typed value.
+     */
+    byteValue(): Uint8Array | null;
+
+    /**
+     * Returns the current value as a [[Decimal]].  This is only valid if `type() == IonTypes.DECIMAL`.
+     *
+     * @return `null` if the current Ion value [[isNull]].
+     *
+     * @throw Error when the reader is not positioned on a `decimal` typed value.
+     */
+    decimalValue(): Decimal | null;
+
+    /**
+     * Returns the current value as a `number`.  This is only valid if `type() == IonTypes.INT`
+     * or `type() == IonTypes.FLOAT`.
+     *
+     * @return `null` if the current Ion value [[isNull]]. For `int` values that are outside of
+     *  the range specified by `Number.MIN_SAFE_INTEGER` and `Number.MAX_SAFE_INTEGER`, this method
+     *  will truncate the result.
+     *
+     * @throw Error when the reader is not positioned on a `int` or `number` typed value.
+     */
+    numberValue(): number | null;
+
+    /**
+     * Returns the current value as a [[Timestamp]].  This is only valid if `type() == IonTypes.TIMESTAMP`.
```

Comment:
this is for `stringValue()`, but appears to be a copy of the timestamp typedoc

Reply:
Whoops--yep